plugins/discourse-local-dates/assets/javascripts/initializers/discourse-local-dates.js.es6

```diff
@@ -1,10 +1,71 @@
+import deprecated from "discourse-common/lib/deprecated";
+import { getOwner } from "discourse-common/lib/get-owner";
+import { hidePopover, showPopover } from "discourse/lib/d-popover";
 import LocalDateBuilder from "../lib/local-date-builder";
-import showModal from "discourse/lib/show-modal";
 import { withPluginApi } from "discourse/lib/plugin-api";
+import showModal from "discourse/lib/show-modal";
+
+export function applyLocalDates(dates, siteSettings) {
+  if (!siteSettings.discourse_local_dates_enabled) {
+    return;
+  }
+
+  const currentUserTZ = moment.tz.guess();
+
+  dates.forEach((element) => {
+    const opts = buildOptionsFromElement(element, siteSettings);
+
+    const localDateBuilder = new LocalDateBuilder(opts, currentUserTZ).build();
+    element.innerText = "";
+    element.insertAdjacentHTML(
+      "beforeend",
+      `
+        <svg class="fa d-icon d-icon-globe-americas svg-icon" xmlns="http://www.w3.org/2000/svg">
+          <use xlink:href="#globe-americas"></use>
+        </svg>
+        <span class="relative-time">${localDateBuilder.formated}</span>
+      `
+    );
+    element.setAttribute("aria-label", localDateBuilder.textPreview);
+
+    const classes = ["cooked-date"];
+    if (localDateBuilder.pastEvent) {
+      classes.push("past");
+    }
+    element.classList.add(...classes);
+  });
+}
+
+function buildOptionsFromElement(element, siteSettings) {
+  const opts = {};
+  const dataset = element.dataset;
+  opts.time = dataset.time;
+  opts.date = dataset.date;
+  opts.recurring = dataset.recurring;
+  opts.timezones = (
+    dataset.timezones ||
+    siteSettings.discourse_local_dates_default_timezones ||
+    "Etc/UTC"
+  )
+    .split("|")
+    .filter(Boolean);
+  opts.timezone = dataset.timezone;
+  opts.calendar = (dataset.calendar || "on") === "on";
+  opts.displayedTimezone = dataset.displayedTimezone;
+  opts.format = dataset.format || (opts.time ? "LLL" : "LL");
+  opts.countdown = dataset.countdown;
+  return opts;
+}
 
 function initializeDiscourseLocalDates(api) {
-  api.decorateCooked(
-    ($elem) => $(".discourse-local-date", $elem).applyLocalDates(),
+  api.decorateCookedElement(
+    (elem) => {
+      const siteSettings = api.container.lookup("site-settings:main");
```

Comment:
Probably a micro-optimization, but should we move this outside of the `api.decorateCookedElement`? That way, we only retrieve the site settings on init rather than everytime we decorate a post?